agent/ecscni/plugin.go

```diff
@@ -111,18 +114,34 @@ func (client *cniClient) setupNS(cfg *Config) (*current.Result, error) {
 	}
 
 	seelog.Debugf("[ECSCNI] Starting ENI (%s) setup in the the container namespace: %s", cfg.ENIID, cfg.ContainerID)
-	os.Setenv("ECS_CNI_LOGLEVEL", logger.GetLevel())
-	defer os.Unsetenv("ECS_CNI_LOGLEVEL")
 
-	// Invoke eni plugin ADD command
-	result, err := client.add(runtimeConfig, cfg, client.createENINetworkConfig)
-	if err != nil {
-		return nil, errors.Wrap(err, "cni setup: invoke eni plugin failed")
+	// Invoke eni plugin ADD command based on the type of eni plugin
+	if cfg.ENIType == apieni.BranchENIType {
+		os.Setenv("VPC_CNI_LOG_LEVEL", logger.GetLevel())
+		os.Setenv("VPC_CNI_LOG_FILE", vpcCNIPluginPath)
+
+		defer os.Unsetenv("VPC_CNI_LOG_LEVEL")
+		defer os.Unsetenv("VPC_CNI_LOG_FILE")
+
+		result, err := client.add(runtimeConfig, cfg, client.createBranchENINetworkConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "cni setup: invoke branch eni plugin failed")
+		}
+		seelog.Debugf("[ECSVPCCNI] Branch ENI setup done: %s", result.String())
+	} else {
+		os.Setenv("VPC_CNI_LOG_LEVEL", logger.GetLevel())
```

Comment:
is client.createENINetworkConfig invoking previous ecs-cni plugin? why not set ECS_CNI_LOGLEVEL?

Reply:
Fixed.